superset/viz.py

```diff
@@ -1532,7 +1532,31 @@ def get_data(self, df: pd.DataFrame) -> VizData:
 
         slice_ids = self.form_data.get("deck_slices")
         slices = db.session.query(Slice).filter(Slice.id.in_(slice_ids)).all()
+
+        features: dict[str, list[Any]] = {}
```

Comment:
noting that we're trying to deprecate `viz.py` in favor of `/api/v1/chart/data`, though might be out-of-scope for this PR. I think some visualization like the deck.gl ones will need to migrate to the chart/data endpoint eventually. Pilling on more stuff onto viz.py might make it harder to migrate.

Reply:
Given this, it might inform doing this in JS if that's possible, and possibly hitting `/api/chart/data` N times (for each chart), and joining the data client-side.